src/test/suite/connectionController.test.ts

```diff
@@ -189,32 +180,45 @@ suite('Connection Controller Test Suite', function () {
     expect(wasSetToConnectingWhenDisconnecting).to.be.true;
   });
 
-  test('"connect()" should fire 3 CONNECTIONS_DID_CHANGE event', (done) => {
-    let eventCounter = 0;
+  test('"connect()" should fire the connections did change event the expected number of types', async () => {
+    // The number of times we expect to re-render connections on the sidebar:
+    // - connection attempt started
+    // - connection attempt finished
+    const expectedTimesToFire = 2;
+    let connectionsDidChangeEventFiredCount = 0;
+
     testConnectionController.addEventListener(
       DataServiceEventTypes.CONNECTIONS_DID_CHANGE,
       () => {
-        eventCounter++;
-        if (eventCounter === 3) {
-          done();
-        }
+        connectionsDidChangeEventFiredCount++;
       }
     );
 
-    void testConnectionController.addNewConnectionStringAndConnect(
+    await testConnectionController.addNewConnectionStringAndConnect(
       TEST_DATABASE_URI
     );
-  });
+    await sleep(100);
```

Comment:
Is there a way we can avoid sleeping an arbitrary time here? Maybe we can poll/wait for the success state?

I added a bunch of `sleep`s in tests in the past and overtime they can really slow down how fast all tests are run, and they also can introduce flakes from slow networks and things, so I've been avoiding them now.

Reply:
Removed as answered below.